console/backend/src/test/java/org/frankframework/management/web/ConfigurationsEndpointTest.java

```diff
@@ -0,0 +1,136 @@
+package org.frankframework.management.web;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockPart;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+@ContextConfiguration(classes = {WebTestConfiguration.class, ConfigurationsEndpoint.class})
+public class ConfigurationsEndpointTest extends FrankApiTestBase {
+
+	@Test
+	public void getConfigurationXML() throws Exception {
+		testActionAndTopicHeaders("/configurations/name", "CONFIGURATION", "GET");
+	}
+
+	@Test
+	public void reloadConfigurationNoValidAction() throws Exception {
+		mockMvc.perform(MockMvcRequestBuilders.put("/configurations/name")
+				.content("{\"action\": \"abc\"}")
+				.contentType(MediaType.APPLICATION_JSON))
+			.andExpect(MockMvcResultMatchers.status().isBadRequest());
+	}
+
+	@Test
+	public void reloadConfiguration() throws Exception {
+		mockMvc.perform(MockMvcRequestBuilders.put("/configurations/name")
+				.content("{\"action\": \"reload\"}")
+				.contentType(MediaType.APPLICATION_JSON))
+			.andExpect(MockMvcResultMatchers.status().isAccepted())
+			.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON));
+	}
+
+	@Test
+	public void fullReloadNoValidAction() throws Exception {
+		mockMvc.perform(MockMvcRequestBuilders.put("/configurations")
+					.content("{\"action\": \"abc\"}")
+					.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.status().isBadRequest());
+	}
+
+	@Test
+	public void fullReload() throws Exception {
+		mockMvc.perform(MockMvcRequestBuilders.put("/configurations")
+					.content("{\"action\": \"reload\"}")
+					.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.status().isAccepted());
+	}
+
+	@Test
+	public void getConfigurationByName() throws Exception {
+		testActionAndTopicHeaders("/configurations/name", "CONFIGURATION", "GET");
+	}
+
+	@Test
+	public void getConfigurationHealth() throws Exception {
+		testActionAndTopicHeaders("/configurations/name/health", "HEALTH", null);
+	}
+
+	@Test
+	public void getConfigurationFlow() throws Exception {
+		testActionAndTopicHeaders("/configurations/name/flow", "FLOW", null);
+	}
+
+	@Test
+	public void getConfigurationDetailsByName() throws Exception {
+		testActionAndTopicHeaders("/configurations/name/versions", "CONFIGURATION", "FIND");
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings = {"{\"activate\":true}", "{\"autoreload\":true}"})
```

Comment:
Afhandeling hier en in de method hieronder zijn voor twee opties (activate/autoreload) precies hetzelfde, daarom parameterized gemaakt.

Reply:
Misschien dan wel een check maken dat hij de juiste header zet